src/sframe/parallel_csv_parser.cpp

```diff
@@ -28,6 +28,20 @@ namespace turi {
 
 using fileio::file_status;
 
+/**
+ * Escape BOMs
+ */
+void skip_BOM(general_ifstream& fin) {
+   char fChar, sChar, tChar;
+   fChar = fin.get();
+   sChar = fin.get();
+   tChar = fin.get();
+   bool isBOM = ((fChar == (char)0xEF) && (sChar == (char)0xBB) && (tChar == (char)0xBF));
+   if (!isBOM) {
+       fin.seekg(0);
```

Comment:
Apparently, I can't do this operation on a `tar.gz` file.

Reply:
Can you do something like seekg(-3) to go back 3 chars?

Reply:
Hmm... the general_fstream does not support seek for compressed files. There is an fstream put back mechanism.

Reply:
This might work http://www.cplusplus.com/reference/istream/istream/putback/